sridha2412/assignment2/q1.py

```diff
@@ -2,29 +2,41 @@
 
 # node class`
 class Node:
-    # constuctor to initlalize a node object
-    def __init__(self, data):
+    def __init__(self, data: int):
+        """ constuctor to initlalize a node object """
         self.data = data
         self.left = None
         self.right = None
 
-# if key is present, print ancestors else return false
-def print_ancestors(root, key):
-    # base case
+def find_ancestors(root: Node, key: int, ancestors: list) -> bool:
+    """ if key is present, return list of ancestors else return false """
+    # base cases
     if root is None:
         return False
-
     if root.data == key:
         return True
 
+    # store all nodes in list
+    ancestors.append(root.data)
```

Comment:
Just note that the exercice request that you do not store additional nodes and values, and list.append is storing the values here :)

Reply:
I thought that constraint was for the second question. 😅  I can't figure out a way to return a list without storing the values, the only way  I can think of is using iterators but that would completely refactor my code. Do you have any other suggestions or do I fully refactor the code?